config/environment/ui-test.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+return array(
+
+    'Piwik\Config' => \DI\decorate(function (\Piwik\Config $config) {
+        $config->General['cors_domains'][] = '*';
+        $config->trusted_hosts[] = $config->tests['http_host'];
+        $config->trusted_hosts[] = $config->tests['http_host'] . ':' . $config->tests['port'];
+        return $config;
+    }),
+
+    'observers.global' => \DI\add(array(
+
+        array('Request.dispatch.end', function (&$result) {
+            $request = $_GET + $_POST;
+
+            // Overlay needs the full URLs in order to find the links in the embedded page (otherwise the %
+            // tooltips don't show up)
+            if (!empty($request['method'])
+                && $request['method'] == 'Overlay.getFollowingPages'
```

Comment:
I can add a ui-test.php file to Overlay. I can't add the event there though, since we'd need to abort the event in config/ui-test.php with the event in Overlay. I could create a test class UITestResponseManipulator that is used in this global event and make it use a blacklist. Then in Overlay, the ui-test.php could just add to the blacklist. Do you think this is a good idea?